### PR TITLE
Disable connect button when connecting.

### PR DESCRIPTION
**Pull request by obraliar**
Issue: https://github.com/JabRef/jabref/issues/1703.

> "open remote database dialog": "Connect" button should be disabled an label change to "Connecting...". Reason: Connection may take longer and pressing the button multiple times leads to error message dialogs. (You know, http://www.freemysqlhosting.net/ latency)

- [x] Manually tested changed features in running JabRef

### CODE REVIEW

**Comment by boceckts** [comment]
LGTM 👍

File: src/main/java/net/sf/jabref/gui/shared/OpenSharedDatabaseDialog.java [hunk 1]
```diff
@@ -90,13 +90,16 @@ public OpenSharedDatabaseDialog(JabRefFrame frame) {
...
 
     public void openSharedDatabase() {
+        setLoadingConnectButton(true);
+
         try {
             bibDatabaseContext.getDBSynchronizer().openSharedDatabase(connectionProperties);
             frame.addTab(bibDatabaseContext, true);
             setGlobalPrefs();
             bibDatabaseContext.getDBSynchronizer().registerListener(new SharedDatabaseUIManager(frame));
             frame.output(Localization.lang("Connection_to_%0_server_stablished.", connectionProperties.getType().toString()));
```
Comment (oscargus) [inline src/main/java/net/sf/jabref/gui/shared/OpenSharedDatabaseDialog.java]: Sorry, but I couldn't help to notice "stablished" which should be "established".

**Comment by oscargus** [comment]
Looks good! Had to think a bit to understand why you return before getting the button back to normal. :-)

File: src/main/java/net/sf/jabref/gui/shared/OpenSharedDatabaseDialog.java (hunk 1 above)
Comment (obraliar) [inline src/main/java/net/sf/jabref/gui/shared/OpenSharedDatabaseDialog.java]: Oh, yes. Done.

File: src/main/resources/l10n/JabRef_de.properties [hunk 2]
```diff
@@ -2473,3 +2473,6 @@ Entered_database_has_obsolete_structure_and_is_no_longer_supported.=Die_gewählt
 However,_a_new_database_was_created_alongside_the_pre-3.6_one.=Dennoch_wurde_eine_neue_Datenbank_neben_der_alten_Datenbank_erzeugt.
 
 Click_here_to_learn_about_the_migration_of_pre-3.6_databases.=Klicken_um_Informationen_zur_Migration_von_vor-3.6-Datenbanken_zu_erhalten.
+Click_here_to_learn_about_the_migration_of_pre-3.6_databases.=
```
Comment (koppor) [inline src/main/resources/l10n/JabRef_de.properties]: Remove this line

File: src/main/java/net/sf/jabref/gui/shared/OpenSharedDatabaseDialog.java [hunk 3]
```diff
@@ -90,13 +90,16 @@ public OpenSharedDatabaseDialog(JabRefFrame frame) {
...
+
         try {
             bibDatabaseContext.getDBSynchronizer().openSharedDatabase(connectionProperties);
             frame.addTab(bibDatabaseContext, true);
             setGlobalPrefs();
             bibDatabaseContext.getDBSynchronizer().registerListener(new SharedDatabaseUIManager(frame));
-            frame.output(Localization.lang("Connection_to_%0_server_stablished.", connectionProperties.getType().toString()));
+            frame.output(Localization.lang("Connection_to_%0_server_established.", connectionProperties.getType().toString()));
             dispose();
+            return;
```
Comment (koppor) [inline src/main/java/net/sf/jabref/gui/shared/OpenSharedDatabaseDialog.java]: Add a comment why you can return here. This was also @oscargus comment.

File: src/main/java/net/sf/jabref/gui/shared/OpenSharedDatabaseDialog.java [hunk 4]
```diff
@@ -329,4 +335,16 @@ private void checkFields() throws JabRefException {
         }
     }
 
+    /**
+     * Sets the connectButton according to the current connection state.
+     */
+    private void setLoadingConnectButton(boolean isLoading) {
```
Comment (koppor) [inline src/main/java/net/sf/jabref/gui/shared/OpenSharedDatabaseDialog.java]: Rename to `setLoadingConnectButtonText`

**Comment by koppor** [comment]
Please fix minor comments, resolve conflicts and take care that no duplicate localizations come in. Then, it's good to go.

File: src/main/resources/l10n/JabRef_de.properties (hunk 2 above)
Comment (obraliar) [inline src/main/resources/l10n/JabRef_de.properties]: Done.

File: src/main/java/net/sf/jabref/gui/shared/OpenSharedDatabaseDialog.java (hunk 3 above)
Comment (obraliar) [inline src/main/java/net/sf/jabref/gui/shared/OpenSharedDatabaseDialog.java]: Done.

File: src/main/java/net/sf/jabref/gui/shared/OpenSharedDatabaseDialog.java (hunk 4 above)
Comment (obraliar) [inline src/main/java/net/sf/jabref/gui/shared/OpenSharedDatabaseDialog.java]: Done.

File: src/main/resources/l10n/JabRef_fr.properties [hunk 5]
```diff
@@ -1711,3 +1711,6 @@ Entered_database_has_obsolete_structure_and_is_no_longer_supported.=Cette_base_d
 However,_a_new_database_was_created_alongside_the_pre-3.6_one.=Ainsi,_une_nouvelle_base_a_été_créée_en_parallèle_de_celle_antérieure_à_la_version_3.6.
 
 Click_here_to_learn_about_the_migration_of_pre-3.6_databases.=Cliquer_ici_pour_plus_de_détails_sur_la_migration_des_bases_antérieures_à_la_version_3.6.
+Click_here_to_learn_about_the_migration_of_pre-3.6_databases.=
```
Comment (matthiasgeiger) [inline src/main/resources/l10n/JabRef_fr.properties]: duplicate here

Comment (obraliar) [inline src/main/resources/l10n/JabRef_fr.properties]: Done. (it comes from auto-merge)

**Pull request merged**